File: rabbitmq-producer/src/main/java/cn/zzk/rabbitmqproducter/config/RabbitmqConfig.java
package cn.zzk.rabbitmqproducter.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class RabbitmqConfig {
    @Bean
    public Jackson2JsonMessageConverter producerJackson2MessageConverter(ObjectMapper objectMapper) {
        return new Jackson2JsonMessageConverter(objectMapper);
    }
}

File: test-container/src/main/kotlin/cn/zzk/testcontainer/RabbitMQConfig.kt
package cn.zzk.testcontainer

import org.springframework.amqp.core.Queue
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RabbitMQConfig {

    @Bean
    fun helloQueue(): Queue {
        return Queue("hello", true)
    }

}